### DISCUSSION TITLE
Help Needed with Fine-Tuning Hyperparameters for Training Script

### DISCUSSION THREAD

**Discussion by ep0p** [mindee/doctr#1675] [Q&A]
I’m working on fine-tuning  crnn_vgg16_b recognition model.

I’ve run into a bit of a challenge: my training loss is fluctuating rapidly, and I haven’t been able to stabilize it despite trying a few different setups.

Since I’m still learning the ropes, I’d appreciate any advice you could provide on the following hyperparameters to help improve stability and performance:

1. Number of Epochs: How many epochs would you recommend starting with, given my dataset size?
2. Learning Rate: Any tips on the initial learning rate and strategies for adjusting it during training? 
3. Batch Size: What batch size would you suggest for optimal performance and stability?

Here’s a quick summary of my setup:

Dataset size: 1 million words
Train/Validation split: 80%/20%
Tried Hyperparameters: ```--epochs 20 --lr 0.0001 --batch_size 16```
```--epochs 50 --lr 0.000001 --batch_size 32 ```
```--epochs 20 --lr 0.000001 --batch_size 32 ```

**Comment by ep0p** [comment]
Figured it out eventually.

**Reply by MRawlani** [reply]
hey, curious to know what worked for you

**Reply by ep0p** [reply]
I could not tell you exactly, since its been a while, and i have tried plenty of ocr models since...
but my conclusion in what concerns doctr is that it was not suited to my case; after many fine tunning sessions, i was not able to get a good solution that would ocr my documents properly;
maybe because my training dataset was a very clean one, and i was just adding fake noise ...i don't know
